--- openjdk-jdk11u-jdk-11.0.8-10_adopt/src/hotspot/os/posix/os_posix.cpp	2020-07-15 03:31:54.000000000 +0800
+++ /data/data/com.termux/files/home/jdk11/openjdk-11.0.8/src/hotspot/os/posix/os_posix.cpp	2020-12-25 15:41:00.730205819 +0800
@@ -46,7 +46,12 @@
 #include <sys/utsname.h>
 #include <time.h>
 #include <unistd.h>
+
+#ifdef __ANDROID__
+#include <utmp.h>
+#else
 #include <utmpx.h>
+#endif // __ANDROID__
 
 // Todo: provide a os::get_max_process_id() or similar. Number of processes
 // may have been configured, can be read more accurately from proc fs etc.
@@ -379,6 +384,17 @@
 void os::Posix::print_uptime_info(outputStream* st) {
   int bootsec = -1;
   int currsec = time(NULL);
+
+#ifdef __ANDROID__
+  struct utmp* ent;
+  setutent();
+  while ((ent = getutent())) {
+    if (!strcmp("system boot", ent->ut_line)) {
+      bootsec = ent->ut_tv.tv_sec;
+      break;
+    }
+  }
+#else
   struct utmpx* ent;
   setutxent();
   while ((ent = getutxent())) {
@@ -387,6 +403,7 @@
       break;
     }
   }
+#endif // __ANDROID__
 
   if (bootsec != -1) {
     os::print_dhm(st, "OS uptime:", (long) (currsec-bootsec));
