--- android-11/external/icu/libandroidicu/include/uconfig_local.h	2020-09-18 15:38:23.027412839 +0800
+++ patches/uconfig_local.h	2020-09-18 15:36:30.004068247 +0800
@@ -22,8 +22,10 @@
 #define U_LIB_SUFFIX_C_NAME _android
 #define U_HIDE_DRAFT_API 1
 #define U_HIDE_DEPRECATED_API 1
-#define U_SHOW_CPLUSPLUS_API 0
-#define U_HIDE_INTERNAL_API 1
+
+// macro modify
+#define U_SHOW_CPLUSPLUS_API 1
+#define U_HIDE_INTERNAL_API 0
 
 // Set this flag to allow header-only C++ usages
 #define LIBANDROIDICU_U_SHOW_CPLUSPLUS_API 1
